tests/test_modeling_electra.py

```diff
@@ -344,3 +344,19 @@ def test_model_from_pretrained(self):
         for model_name in ELECTRA_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
             model = ElectraModel.from_pretrained(model_name)
             self.assertIsNotNone(model)
+
+
+@require_torch
+class ElectraModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_no_head_absolute_embedding(self):
+        model = ElectraModel.from_pretrained("google/electra-small-discriminator")
```

Comment:
Could you use a `ElectraForPreTraining` instead? This model can load the discriminator and has a custom head so it would be nice to test it at well.

Reply:
I have tried for ElectraForPreTraining, however hidden state is coming none for it.

Below is code and output.

model = ElectraForPreTraining.from_pretrained("google/electra-small-discriminator")
input_ids = torch.tensor([[0, 345, 232, 328, 740, 140, 1695, 69, 6078, 1588, 2]])
output = model(input_ids)

output 

ElectraForPreTrainingOutput(loss=None, logits=tensor([[-8.3982, -4.4508, -4.5948, -4.6049, -4.5626, -4.5957, -4.0569, -4.7136,-3.4094, -3.1778, -4.8272]], grad_fn=<SqueezeBackward1>), hidden_states=None, attentions=None)

Reply:
Yes, that is the case for `ElectraModel` as well. You didn't put `output_attentions=True`, so it's not outputting the attentions, only the logits (which is what you're evaluating here).

Reply:
As pointed out by @spatil6, I meant `output_hidden_states` not being set to `True` and therefore not outputting the `hidden_states` value, not the `attentions`.